Formula/podman.rb

```diff
@@ -0,0 +1,26 @@
+class Podman < Formula
+  desc "Tool for managing OCI containers and pods"
+  homepage "https://podman.io/"
+  url "https://github.com/containers/podman/archive/v2.0.5.tar.gz"
+  sha256 "cd7618429f02db8e7dbca29c6f8a7623a91a01804ef768dc5ec8c7660679b0c5"
+  license "Apache-2.0"
+
+  depends_on "go-md2man" => :build
+  depends_on "go@1.14" => :build
+
+  def install
+    system "make", "podman-remote-darwin"
+    bin.install "bin/podman-remote-darwin" => "podman"
+
+    inreplace "docs/remote-docs.sh", "sed -i", "sed -i \"\""
+    system "make", "install-podman-remote-darwin-docs"
+    man1.install Dir["docs/build/remote/darwin/*.1"]
+
+    bash_completion.install "completions/bash/podman"
+    zsh_completion.install "completions/zsh/_podman"
+  end
+
+  test do
+    assert_match "podman version #{version}", shell_output("#{bin}/podman -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks so much for your write up on writing tests, I really appreciate maintainers going out of their way to explain something so thoroughly :)

I'm having a little trouble with this. The test I added should be for the last case:
Expected correct behavior: 
```
$ podman 
Error: Get "http://d/v1.0.0/libpod../../../_ping": dial unix ///var/folders/15/5t5gspkd4qq7n9y4smb2g_nh0000gn/T/run--1/podman/podman.sock: connect: no such file or directory
```
I added
```
assert_match "Error: Get \"http://d/v1.0.0/libpod../../../_ping\": dial unix",  shell_output("#{bin}/podman")
```

And the output is 
```
$ brew test podman
==> Testing podman
==> /usr/local/Cellar/podman/2.0.5/bin/podman -v
==> /usr/local/Cellar/podman/2.0.5/bin/podman
Error: Get "http://d/v1.0.0/libpod../../../_ping": dial unix ///private/tmp/run--1/podman/podman.sock: connect: no such file or directory
Error: podman: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<125>.
```

I don't understand how the assertionfailed doesn't say </Error/> expected, and is instead <0>. If you can point me in the right direction, that would be great.

Reply:
`shell_output` takes an optional second parameter for the returncode of the command. If none is de it expects `<0>`, but seeing the expected output it should probably expect something that isn't 0.

Reply:
Thanks so much! Updated :)